src/crawlers/browser_crawler.js

```diff
@@ -194,21 +195,24 @@ class BrowserCrawler extends BasicCrawler {
         request.loadedUrl = await page.url();
     }
 
-    async _createContextFunction(pluginOptions) {
+    async extendLaunchContext(launchContext) {
+        const crawlingContext = this.crawlingContexts.get(launchContext.contextId);
```

Comment:
Hmm, I hope I have not found a problem. The thing is `launchContext` makes sense per browser, whereas `contextId` exists per page. What happens when we keep calling `newPage()` with new IDs but no browsers are launched? Only new pages in existing browsers are created?

Reply:
It will have the new context id but with the old proxy and session.